Source/SwiftLintBuiltInRules/Rules/Style/ImplicitOptionalInitializationRule.swift

```diff
@@ -0,0 +1,160 @@
+import SwiftSyntax
+
+@SwiftSyntaxRule(explicitRewriter: true)
+struct ImplicitOptionalInitializationRule: Rule {
+    var configuration = ImplicitOptionalInitConfiguration()
+
+    static let description = RuleDescription(
+        identifier: "implicit_optional_initialization",
+        name: "Implicit Optional Initialization",
+        description: "Optionals should be consistently initialised",
+        kind: .style,
+        nonTriggeringExamples: ImplicitOptionalInitRuleExamples.nonTriggeringExamples,
+        triggeringExamples: ImplicitOptionalInitRuleExamples.triggeringExamples,
+        corrections: ImplicitOptionalInitRuleExamples.corrections,
+        deprecatedAliases: ["redundant_optional_initialization"]
+    )
+}
+
+private extension ImplicitOptionalInitializationRule {
+    final class Visitor: ViolationsSyntaxVisitor<ConfigurationType> {
+        override func visitPost(_ node: VariableDeclSyntax) {
+            violations.append(
+                contentsOf: node.violationPositions(for: configuration.style).map {
+                    ReasonedRuleViolation(position: $0, reason: reason)
+                })
+        }
+
+        var reason: String {
+            let recommendation =
+                switch configuration.style {
+                case .always: "explicitly initialized to nil"
+                case .never: "implicitly initialized without nil"
+                }
+
+            return "Optionals should be \(recommendation) in variable declarations"
+        }
+    }
+}
+
+private extension ImplicitOptionalInitializationRule {
+    final class Rewriter: ViolationsSyntaxRewriter<ConfigurationType> {
+        override func visit(_ node: VariableDeclSyntax) -> DeclSyntax {
+            let violationPositions = node.violationPositions(for: configuration.style)
+            guard !violationPositions.isEmpty else { return super.visit(node) }
+
+            numberOfCorrections +=
+                violationPositions.filter {
+                    !$0.isContainedIn(
```

Comment:
You wouldn't need this if you were to visit `PatternBindingSyntax` elements here instead of `VariableDeclSyntax`. Construction of the replacement could also be easier that way. Same for the visitor.

With `node.parent` you can access the variable declaration to retrieve the type.